src/components/QueryEditor/VisualQueryEditor/KQLFilter.tsx

```diff
@@ -24,6 +24,10 @@ interface KQLFilterProps {
   onChange: (query: KustoQuery) => void;
 }
 
+export interface FilterExpression extends QueryEditorOperatorExpression {
+  index: number;
+}
```

Comment:
Note for the reviewer: I had to define this new type because `EditorList` from @grafana/ui does not include the index of the element being rendered. My plan is to open an issue / PR there to add that but in the meantime this is necessary to know the possition of the filter (and conditionally render the "OR" label).